### PR TITLE
[refactor] Move Kernel::lower() outside the taichi::lang::Kernel

### PR DESCRIPTION
**Pull request by PGZXB**
Issue: #7002

### CODE REVIEW

File: taichi/transforms/compile_to_offloads.cpp [hunk 1]
```diff
@@ -41,6 +41,14 @@ void compile_to_offloads(IRNode *ir,
   auto print = make_pass_printer(verbose, kernel->get_name(), ir);
   print("Initial IR");
 
+  if (config.print_preprocessed_ir && start_from_ast) {
+    TI_INFO("[{}] {}:", kernel->get_name(), "Preprocessed IR");
+    std::cout << std::flush;
+    irpass::re_id(ir);
+    irpass::print(ir);
+    std::cout << std::flush;
```
Comment (strongoier) [inline taichi/transforms/compile_to_offloads.cpp:49]: Maybe we don't need to print preprocessed IR again when `verbose == true`. WDYT @lin-hitonami

Comment (PGZXB) [reply taichi/transforms/compile_to_offloads.cpp:49]: Agree!

Comment (lin-hitonami) [reply taichi/transforms/compile_to_offloads.cpp:49]: makes sense.

**Review by strongoier — APPROVED** [review]

**Pull request merged**